### PR TITLE
validate commit has a jira issue

### PR DESCRIPTION
**Pull request by jctanner**
Per https://issues.redhat.com/browse/AAH-1106 all commits must have a JIRA issue and every commit must have a CHANGELOG entry.

### CODE REVIEW

**Comment by awcrosby** [comment]
When we added the `galaxy_ng` `validate_commit_message_custom.py` to `galaxy-importer` - in order to automate the need to remember to keep them in sync, I added a comparison script - if you think that is useful it can be done here: 
https://github.com/ansible/galaxy-importer/blob/master/.github/workflows/galaxy_ng_commit_check.yml#L25-L35

**Comment by jctanner** [comment]
@awcrosby we could also simply wget the script from the galaxy_ng repo instead of having to maintain it in multiple spots. What do you think about that?

**Comment by awcrosby** [comment]
@jctanner yeah that sounds good!

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -5,6 +5,33 @@ on:
...
+      - name: Run script to validate commits for both pull request and a push
+        env:
+          PY_COLORS: '1' 
+          ANSIBLE_FORCE_COLOR: '1' 
+          GITHUB_PULL_REQUEST: ${{ github.event.number }}
+          GITHUB_PULL_REQUEST_BODY: ${{ github.event.pull_request.body }}
+          GITHUB_BRANCH: ${{ github.head_ref }}
+          GITHUB_REPO_SLUG: ${{ github.repository }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_CONTEXT: ${{ github.event.pull_request.commits_url }}
```
Comment (himdel) [inline .github/workflows/pr-checks.yml:28]: none of these env vars seem to be used? I'm only seeing references to `GITHUB_PR_COMMITS_URL`, `START_COMMIT` and `END_COMMIT` in the code

File: .github/workflows/pr-checks.yml [hunk 2]
```diff
@@ -5,6 +5,33 @@ on:
...
+          GITHUB_PR_COMMITS_URL: ${{ github.event.pull_request.commits_url }}
+          START_COMMIT: ${{ github.event.before }}
+          END_COMMIT: ${{ github.event.after }}
+        run: |
+          curl https://raw.githubusercontent.com/ansible/galaxy_ng/master/.ci/scripts/validate_commit_message_custom.py | python
```
Comment (himdel) [inline .github/workflows/pr-checks.yml:33]: this needs addiitional logic to ignore pull requests made by dependabot , those will not have `Fixes:` nor `No-Issue`

Comment (himdel) [reply .github/workflows/pr-checks.yml:33]: And I think we also need the same treatmeant for patchback .. example https://github.com/ansible/ansible-hub-ui/pull/1369 ... the PR does link to the issue, but the commit message doesn't

Comment (himdel) [reply .github/workflows/pr-checks.yml:33]: I think something like

```
if: ${{ github.event.pull_request.user.login }} != "dependabot[bot]" && ${{ github.event.pull_request.user.login }} != "patchback[bot]"
```

should fix it, I think :)

**Comment by himdel** [comment]
hm, @awcrosby I think the script has a bug now - it requires the commit with `Fixes: AAH-123" and the one with `CHANGES/123.misc` to be the same commit *even* in PR checks.

Can we fix that? They become a single commit anyway on squash, so let's allow the message in separate commits.

File: .github/workflows/pr-checks.yml (hunk 2 above)
Comment (jctanner) [reply .github/workflows/pr-checks.yml:26]: https://github.com/ansible/galaxy_ng/pull/1088

**Comment by jctanner** [comment]
waiting for https://github.com/ansible/galaxy_ng/pull/1088 to get merged.

**Review by himdel — APPROVED** [review]
https://github.com/ansible/galaxy_ng/pull/1088 is in,
I think this is ready :+1: :)

**Pull request merged**